File: web/src/lib/stores/transcription.ts
import { writable } from 'svelte/store';

export type TranscriptionSegment = {
  start: number;
  end: number;
  text_hy: string;
  text_en?: string;
};

export type TranscriptionJob = {
  job_id: string;
  status: string;
  language: string;
  segments: TranscriptionSegment[];
  translated: boolean;
};

export const job = writable<TranscriptionJob | null>(null);

File: app/schemas/transcription.py
"""Transcription response models."""
from datetime import timedelta
from pydantic import BaseModel


class TranscriptionSegment(BaseModel):
    start: float
    end: float
    text_hy: str
    text_en: str | None = None

    @property
    def duration(self) -> float:
        return self.end - self.start


class TranscriptionJobResponse(BaseModel):
    job_id: str
    status: str = "queued"
    language: str = "hy"
    segments: list[TranscriptionSegment] = []
    translated: bool = False

    @property
    def total_duration(self) -> float:
        return sum(segment.duration for segment in self.segments)

    @property
    def total_segments(self) -> int:
        return len(self.segments)
